### ISSUE TITLE
How long time will TSP training take?

### ISSUE THREAD

**Issue by softgearko** [yd-kwon/POMO#1]
Hello.
Thank you for your contribution.
I'm running TSP training with default setting. It took 14 days already and still learning. I wonder how long time this TSP training takes. I'm using V100-DGX station with cuda 10.1.

**Comment by yd-kwon** [comment]
Hi. Thank you so much for your interest in our code!! 

First of all, I am very sorry that our code did not finish in time and make you wait for 14 days and more. This certainly is a very very very long time for something that is not so practical anyway.

Maybe I should change the default setting to training 20 node TSP. I assume most people who are running the codes are just curious how the training works. They wouldn’t care whether it is for 20 node TSP or 100 node TSP.

When we were running the experiments, it took less than 2 weeks to get to 2000 epoch. How have epochs have you run in 14 days? 
Maybe the default setting we uploaded on github is not the original as we have hoped? Some mistakes are possible, as we have trimmed the codes here and there before we made them public. I will double-check, test the codes again, and get back to you later.

In the mean time, if your interest is in creating a well-trained model that shows good final performance, I recommend manual one-time learning rate decay by a factor of 1/10 or 1/100. 
That is, if the log indicates that the training curve has more or less converged, stop training, save the model, and continue training with much smaller learning rates for a few more epochs. This definitely helps in boosting the model’s performance for one last time.
(You can easily get better result than what is reported in the POMO paper this way.)

**Comment by softgearko** [comment]
Thank you for your reply. Anyway, My training was finished after almost 15 days. 
Inference speed seems excellent.

**Comment by yd-kwon** [comment]
I am closing this issue.  In a day or two, I will upload a .py based version of POMO code that one can easily run on servers, without jupyter_notebook applications. The default setting will be N=20.

I have tested code for CVRP100 problems. It took 12 days to train a good model on one V100 GPU. 
![latest-train_score](https://user-images.githubusercontent.com/54298772/124439788-cf7f2480-ddb4-11eb-8365-fc0e6d1e5e73.jpg)

For TSP100, it should take much shorter time.
